x-pack/legacy/plugins/siem/public/pages/detection_engine/create_rule/components/step_define_rule/schema.tsx

```diff
@@ -0,0 +1,146 @@
+    label: i18n.translate(
+      'xpack.siem.detectionEngine.createRule.stepDefineRule.fieldOutputIndiceNameLabel',
+      {
+        defaultMessage: 'Output index name',
+      }
+    ),
+    validations: [
+      {
+        validator: emptyField(
+          i18n.translate(
+            'xpack.siem.detectionEngine.createRule.stepDefineRule.outputIndiceNameFieldRequiredError',
+            {
+              defaultMessage: 'An output indice name for signals is required.',
+            }
+          )
+        ),
+      },
+    ],
+  },
+  useIndicesConfig: {
+    type: FIELD_TYPES.RADIO_GROUP,
+    label: i18n.translate(
+      'xpack.siem.detectionEngine.createRule.stepDefineRule.fieldIndicesTypeLabel',
+      {
+        defaultMessage: 'Indices type',
+      }
+    ),
+    options: [
+      {
+        id: 'true',
+        label: i18n.translate(
+          'xpack.siem.detectionEngine.createRule.stepDefineRule.indicesFromConfigDescription',
+          {
+            defaultMessage: 'Use Elasticsearch indices from SIEM advanced settings',
+          }
+        ),
+      },
+      {
+        id: 'false',
+        label: i18n.translate(
+          'xpack.siem.detectionEngine.createRule.stepDefineRule.indicesCustomDescription',
+          {
+            defaultMessage: 'Provide custom list of indices',
+          }
+        ),
+      },
+    ],
+  },
+  index: {
+    type: FIELD_TYPES.COMBO_BOX,
+    label: i18n.translate('xpack.siem.detectionEngine.createRule.stepAboutRule.fieldTagsLabel', {
+      defaultMessage: 'Indices',
+    }),
+    labelAppend: globalI18n.OPTIONAL_FIELD,
+    validations: [
+      {
+        validator: emptyField(
+          i18n.translate(
+            'xpack.siem.detectionEngine.createRule.stepDefineRule.outputIndiceNameFieldRequiredError',
+            {
+              defaultMessage: 'An output indice name for signals is required.',
+            }
+          )
+        ),
+      },
+    ],
+  },
+  queryBar: {
+    label: i18n.translate(
+      'xpack.siem.detectionEngine.createRule.stepDefineRule.fieldQuerBarLabel',
+      {
+        defaultMessage: 'Custom query',
+      }
+    ),
+    validations: [
+      {
+        validator: (
+          ...args: Parameters<ValidationFunc>
+        ): // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        ReturnType<ValidationFunc<any, ERROR_CODE>> | undefined => {
+          const [{ value, path }] = args;
+          const { query, filters } = value as FieldValueQueryBar;
+          return isEmpty(query.query as string) && isEmpty(filters)
+            ? {
+                code: 'ERR_FIELD_MISSING',
+                path,
+                message: CUSTOM_QUERY_REQUIRED,
+              }
+            : undefined;
+        },
+      },
+      {
+        validator: (
+          ...args: Parameters<ValidationFunc>
+        ): // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        ReturnType<ValidationFunc<any, ERROR_CODE>> | undefined => {
```

Comment:
Optional: You can use this: `ReturnType<ValidationFunc<{}, ERROR_CODE>> | undefined => {`

and you won't need the any and linter disable